src/components/Images/Images.js

```diff
@@ -1,64 +1,347 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import * as Cloudinary from 'cloudinary-react'
+import cloudinary from 'cloudinary-core'
+import useRequired from '../../hooks/useRequired.js'
+import useInvalid from '../../hooks/useInvalid.js'
+import { Media } from '../Media/Media'
+import lazysizes from 'lazysizes'
 
 // https://cloudinary.com/documentation/image_transformation_reference
 
 export const CLOUDINARY_CLOUD_NAME = 'getethos'
 
-const publicIdBase = `https://res.cloudinary.com/${CLOUDINARY_CLOUD_NAME}/image/upload/`
+const cld = new cloudinary.Cloudinary({
+  cloud_name: CLOUDINARY_CLOUD_NAME,
+})
 
-export const CloudinaryImage = ({ publicId, className, crop, ...rest }) => {
+export const CloudinaryImage = ({
+  publicId,
+  className,
+  alt,
+  width,
+  height,
+  crop,
+  ...rest
+}) => {
+  // Verify that all required props were supplied
+  const [includesRequired] = useRequired(['publicId', 'alt'])
+  let allRelevantProps = Object.assign({}, rest, {
+    publicId,
+    alt,
+    width,
+    height,
+  })
+  includesRequired(allRelevantProps)
+
+  // Verify that no invalid props were supplied
+  const [includesInvalid] = useInvalid(
+    Object.keys(CloudinaryImage.PUBLIC_PROPS)
+  )
+  includesInvalid(rest)
+
+  const publicIdBase = `https://res.cloudinary.com/${CLOUDINARY_CLOUD_NAME}/image/upload/`
   const publicIdFilename = publicId.replace(publicIdBase, '')
+
+  // Serve a simpler version if resource is SVG
+  if(publicId.split('.').pop() === 'svg'){
+
+    const baseSvgSettings = {
+      secure: true,
+    }
+
+    const svgUrl = cld.url(publicIdFilename, baseSvgSettings)
+
+    return (
+      <img
+        data-src={svgUrl}
+        className={['Image lazyload', className].join(' ')}
+        alt={alt}
+      />
+    )
+  }
+
+  const baseImageSettings = {
+    quality: 'auto:eco',
+    crop: crop,
+    secure: true,
+    flags: ['progressive:semi'],
+  }
+
+  const dpr1Setting = {
+    dpr: '1.0'
+  }
+
+  const dpr2Setting = {
+    dpr: '2.0'
+  }
+
+  const dpr3Setting = {
+    dpr: '3.0'
+  }
+
+  // ------------------------------
```

Comment:
These settings should probably be rolled up into a mapping of the 3 DPR settings nested within the file format and media breakpoint settings, to write out each `<source>` markup. Then we don't have the repeated constant assignments and instead something like:
```
// for each file type (webp/jp2/jpeg) output each media breakpoint as
<source media="{[arrayOfDprVariations ]}"/>

And just have to do an exception or separate output for the last tag, `<img>`, within `<picture>`. That should DRY up this code a bunch.

Reply:
yeah, I'd love to see it streamlined somehow so if ^ idea does that I'm all for it 👍